core/controllers/acl_decorators.py

```diff
@@ -2933,3 +2933,82 @@ def test_request_originates_from_valid_oppia_ml_instance(self, **kwargs):
+
+
+def can_update_suggestions(handler):
+    """Decorator to check whether the current user can update suggestions.
+
+    Args:
+        handler: function. The function to be decorated.
+
+    Returns:
+        function. The newly decorated function that has common checks and
+        permissions specified by passed in decorator. This function is allowed
+        to be called for users who have the rights to accept suggestions and
+        the suggesters who made their own suggestions.
+
+    Raises:
+        NotLoggedInException. The user is not logged in.
+        UnauthorizedUserException. The user does not have credentials to
+            edit this suggestion.
+        InvalidInputException. The submitted suggestion id is not valid.
+        PageNotFoundException. A suggestion is not found with the given
+            suggestion id.
+    """
+    def test_can_update_suggestion(
+            self, suggestion_id, **kwargs):
+        """Returns a handler to test whether a
+        suggestion can be updated based on the user's roles.
+
+        Args:
+            suggestion_id: str. The suggestion id.
+            **kwargs: *. Keyword arguments.
+
+        Returns:
+            function. The handler for updating a suggestion.
+
+        Raises:
+            NotLoggedInException. The user is not logged in.
+            UnauthorizedUserException. The user does not have credentials to
+                edit this suggestion.
+            InvalidInputException. The submitted suggestion id is not valid.
+            PageNotFoundException. A suggestion is not found with the given
+                suggestion id.
+        """
+        if not self.user_id:
+            raise base.UserFacingExceptions.NotLoggedInException
+        user_actions = self.user.actions
+
+        if len(suggestion_id.split('.')) != 3:
+            raise self.InvalidInputException(
+                'Invalid format for suggestion_id.'
+                ' It must contain 3 parts separated by \'.\'')
+
+        suggestion = suggestion_services.get_suggestion_by_id(suggestion_id)
+
+        if suggestion is None:
+            raise self.PageNotFoundException
+
+        # TODO(#11735): Currently, only checks whether users can review
+        # translation sugeestions. Another check should be added to check
+        # whether users can review question suggestions if the suggestion
+        # type is add question.
+        if suggestion.suggestion_type == (
+                feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT):
+            if user_services.can_review_translation_suggestions(
+                    self.user_id,
+                    language_code=suggestion.change.language_code):
+                return handler(self, suggestion_id, **kwargs)
+
+            if role_services.ACTION_ACCEPT_ANY_SUGGESTION in user_actions:
+                return handler(self, suggestion_id, **kwargs)
+
+            if suggestion_services.check_can_resubmit_suggestion(
```

Comment:
Are we expecting the author to update the suggestion? Also, it's getting a bit confusing with terms like resubmit/update etc. Can we do the direct check here i.e, suggestion.author_id == self.user_id?

Reply:
I agree with @DubeySandeep on this -- I think it makes sense to inline the check here and get rid of the extra function. You already have the suggestion object available here anyway.